### PR TITLE
require composer-packages at launch

### PR DESCRIPTION
**Pull request by sophiewigmore**
## Summary

Similarly to https://github.com/paketo-buildpacks/php-start/issues/16, this buildpack should require `composer-packages` in instances where a `composer.json` or the `$COMPOSER` environment variable are set.

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by sophiewigmore** [comment]
@thitch97 minor release since the API is changing

File: detect_test.go [hunk 1]
```diff
@@ -79,4 +158,25 @@ func testDetect(t *testing.T, context spec.G, it spec.S) {
...
+				Expect(os.Chmod(workingDir, 0000)).To(Succeed())
+				Expect(os.Setenv("COMPOSER", filepath.Join(workingDir, "other-file.json"))).To(Succeed())
+			})
+
+			it.After(func() {
+				Expect(os.Unsetenv("COMPOSER")).To(Succeed())
+				Expect(os.Chmod(workingDir, os.ModePerm)).To(Succeed())
+			})
+
+			it("detection passes, and requires composer-packages", func() {
```
Comment (joshuatcasey) [inline detect_test.go:174]: Is this the right description?

File: detect_test.go [hunk 2]
```diff
@@ -79,4 +158,25 @@ func testDetect(t *testing.T, context spec.G, it spec.S) {
 			})
 		})
 	})
+
+	context("failure cases", func() {
+		context("$COMPOSER is set to a non-existent file", func() {
+			it.Before(func() {
+				Expect(os.Chmod(workingDir, 0000)).To(Succeed())
```
Comment (joshuatcasey) [inline detect_test.go:165]: Do we expect this to happen? Would the lifecycle give us a file with this permission?

File: detect_test.go (hunk 1 above)
Comment (sophiewigmore) [reply detect_test.go:174]: copy paste error!

**Comment by sophiewigmore** [comment]
bump @thitch97

**Review by thitch97 — APPROVED** [review]

**Pull request merged**